### PR TITLE
[DRUP-554] No anonymous users.

### PR DESCRIPTION
**Pull request by Jaesin**
This prevents admins from allowing anonymous users access to monetization.

### CODE REVIEW

File: src/Monetization.php [hunk 1]
```diff
@@ -406,4 +420,49 @@ public function getPrepaidBalanceReports(string $developer_id, \DateTimeImmutabl
...
+      }
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function userRolePresave(RoleInterface $user_role) {
+    // This prevents monetization permission grants on config import.
+    if ($user_role->id() === 'anonymous') {
```
Comment (shadcn) [inline src/Monetization.php]: Suggestion: `AccountInterface::ANONYMOUS_ROLE`

File: src/Monetization.php [hunk 2]
```diff
@@ -406,4 +420,49 @@ public function getPrepaidBalanceReports(string $developer_id, \DateTimeImmutabl
...
+   * Gets a list of Apigee monetization permissions.
+   *
+   * These permissions should only be applied to authenticated roles.
+   *
+   * @return array
+   *   Permissions for the monetization module.
+   */
+  protected function getMonetizationPermissions() {
+    return array_filter($this->permission_handler->getPermissions(), function ($permission) {
+      return ($permission['provider'] === 'apigee_m10n');
```
Comment (shadcn) [inline src/Monetization.php:464]: Do we need to handle m10n submodules? I know we're going to have a few permissions in `apigee_m10n_add_credit`.

**Review by shadcn — COMMENTED** [review]
@Jaesin I added a few comments.

File: src/Monetization.php (hunk 2 above)
Comment (Jaesin) [reply src/Monetization.php:464]: Yes, but those will be follow-ups to this.

File: src/Monetization.php (hunk 1 above)
Comment (Jaesin) [reply src/Monetization.php]: Good Idea

**Review by shadcn — APPROVED** [review]
Approved

**Pull request merged**